addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1028,6 +1028,90 @@ registry.HeaderHamburgerFull = publicWidget.Widget.extend({
     },
 });
 
+registry.BottomFixedElement = publicWidget.Widget.extend({
+    selector: '#wrapwrap',
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    start() {
+        this.$target.on('scroll.bottom_fixed_element', _.debounce(this._hideBottomFixedElements.bind(this), 100));
+        $(window).on('resize.bottom_fixed_element', _.debounce(this._hideBottomFixedElements.bind(this), 100));
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this._super(...arguments);
+        this.$target.off('.bottom_fixed_element');
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Hide the element that is fixed at the bottom of the screen if the scroll
+     * reaches the bottom of the page and if this element hides a button.
+     *
+     * @private
+     */
+    _hideBottomFixedElements() {
+        const self = this;
+        const $bottomFixedElements = $('.o_bottom_fixed_element');
+        if ($bottomFixedElements.length) {
```

Comment:
Useless check

Reply:
why useless ? we don't need to check if we have reached the end of the page if no overlay element

Reply:
In general, I prefer less or more readable code over micro-optimisation. Here we are talking about avoiding to check if we have reached the end of the page **once**, that's like a micro second (and searching the DOM with jQuery just above was an operation which costs already like a 100times more than that). On top of that, you are reducing performance (negligible too) when there are overlay elements in the page because of the extra check.

But it is a matter of taste I guess, so if you want to keep the check, this is good too :wink: